### DISCUSSION TITLE
lotus-miner sectors remove / lotus-miner sectors update-state don't work

### DISCUSSION THREAD

**Discussion by cryptowhizzard** [filecoin-project/lotus#5650] [Storage Provider] [answered]
**Describe the bug**
lotus-miner sectors remove / lotus-miner sectors update-state don't work. 

**Version (run `lotus version`):**
V1.4.2

**Additional context**
Both commands don't work to remove sectors if the files do not exist anymore on disk?

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
that is expected - how to remove them if they aren't there?

try creating mock versions of the sectors in the sealing folders and run the remove again. a simpel miner restart might do the job too

and the version is 1.4.2-rc1 :P

**Comment by cryptowhizzard** [comment]
Well .... lotus-miner sectors list still shows them. 

All these sectors below do not respond to anything.

ID    State           OnChain  Active  Expiration                     Deals  DealWeight  VerifiedPower  
17    removing        NO       NO      n/a                            CC                                
40    Precommit2      NO       NO      n/a                            CC                                
5188  FinalizeSector  YES      NO      1588722 (in 1 year 1 week)     CC                                
5189  FinalizeSector  YES      NO      1588722 (in 1 year 1 week)     CC                                
5423  FinalizeSector  YES      NO      1634802 (in 1 year 4 weeks)    CC                                
5424  FinalizeSector  YES      NO      1634802 (in 1 year 4 weeks)    CC                                
5434  FinalizeSector  YES      NO      1634802 (in 1 year 4 weeks)    CC                                
5496  FinalizeSector  YES      NO      1646322 (in 1 year 4 weeks)    CC                                
5715  Committing      NO       NO      n/a                            CC                                
5716  Committing      NO       NO      n/a                            CC                                
5720  Committing      NO       NO      n/a                            CC                                
5721  Committing      NO       NO      n/a                            CC                                
5724  Committing      NO       NO      n/a                            CC                                
5725  Committing      NO       NO      n/a                            CC                                
5726  Committing      NO       NO      n/a                            CC                                
5727  Committing      NO       NO      n/a                            CC                                
5767  Committing      NO       NO      n/a                            CC                                
5797  Committing      NO       NO      n/a                            CC                                
6345  Committing      NO       NO      n/a                            CC                                
6346  Committing      NO       NO      n/a                            CC                                
6351  Committing      NO       NO      n/a                            CC                                
6353  Committing      NO       NO      n/a                            9                                 
6355  Committing      NO       NO      n/a                            CC                                
6360  Committing      NO       NO      n/a                            CC                                
6362  Committing      NO       NO      n/a                            CC                                
6820  FinalizeSector  YES      NO      1807602 (in 1 year 12 weeks)   CC                                
7007  Committing      NO       NO      n/a                            CC                                
7699  FinalizeSector  YES      NO      1896882 (in 1 year 17 weeks)   CC                                
7702  FinalizeSector  YES      NO      1896882 (in 1 year 17 weeks)   CC                                
7704  FinalizeSector  YES      NO      1896882 (in 1 year 17 weeks)   CC                                
7705  FinalizeSector  YES      NO      1896882 (in 1 year 17 weeks)   CC                                
7706  FinalizeSector  YES      NO      1896882 (in 1 year 17 weeks)   CC                                
7707  FinalizeSector  YES      NO      1896882 (in 1 year 17 weeks)   CC                                
7709  Committing      NO       NO      n/a                            CC                                
7710  Committing      NO       NO      n/a                            CC                                
7711  Committing      NO       NO      n/a                            CC                                
7712  Committing      NO       NO      n/a                            CC    

Time to clean?

Btw ... making dummy sectors did not help.

**Comment by utgarda** [comment]
Tried removing 2 sectors in `PreCommitFailed` state,
```
lotus-miner sectors remove --really-do-it 22
lotus-miner sectors remove --really-do-it 21
```

```
{"level":"info","ts":"2021-04-30T17:07:48.025+0300","logger":"stores","caller":"stores/local.go:608","msg":"remove /fc/fast/sealing/cache/s-t0713585-22"}
{"level":"info","ts":"2021-04-30T17:07:48.394+0300","logger":"stores","caller":"stores/local.go:608","msg":"remove /fc/fast/sealing/unsealed/s-t0713585-22"}
{"level":"error","ts":"2021-04-30T17:07:48.423+0300","logger":"evtsm","caller":"go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/machine.go:83","msg":"Executing event planner failed: running planner for state Removing failed:\n    github.com/filecoin-project/lotus/extern/storage-sealing.(*Sealing).plan\n        /home/etsvigun/lotus/extern/storage-sealing/fsm.go:250\n  - planner for state Removing received unexpected event sealing.SectorRetryPreCommitWait ({User:{}}):\n    github.com/filecoin-project/lotus/extern/storage-sealing.planOne.func1\n        /home/etsvigun/lotus/extern/storage-sealing/fsm.go:524"}
{"level":"warn","ts":"2021-04-30T17:07:51.589+0300","logger":"rpc","caller":"go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:279","msg":"error in RPC call to 'Filecoin.SectorRemove': normal shutdown of state machine:\n    github.com/filecoin-project/go-statemachine.init\n        /home/etsvigun/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/machine.go:16"}
{"level":"warn","ts":"2021-04-30T17:08:48.003+0300","logger":"sectors","caller":"storage-sealing/fsm.go:509","msg":"sector 16 got error event sealing.SectorCommitFailed: failed to wait for porep inclusion: failed to load message: blockstore: block not found"}
{"level":"warn","ts":"2021-04-30T17:08:48.003+0300","logger":"sectors","caller":"storage-sealing/fsm.go:509","msg":"sector 17 got error event sealing.SectorCommitFailed: failed to wait for porep inclusion: failed to load message: blockstore: block not found"}
{"level":"info","ts":"2021-04-30T17:08:48.003+0300","logger":"stores","caller":"stores/local.go:608","msg":"remove /fc/fast/1/sealed/s-t0713585-21"}
{"level":"info","ts":"2021-04-30T17:08:48.006+0300","logger":"sectors","caller":"storage-sealing/states_failed.go:26","msg":"CommitFailed(16), waiting 59.993832164s before retrying"}
{"level":"info","ts":"2021-04-30T17:08:48.006+0300","logger":"sectors","caller":"storage-sealing/states_failed.go:26","msg":"CommitFailed(17), waiting 59.993099052s before retrying"}
{"level":"info","ts":"2021-04-30T17:08:48.026+0300","logger":"stores","caller":"stores/local.go:608","msg":"remove /fc/fast/1/cache/s-t0713585-21"}
{"level":"info","ts":"2021-04-30T17:08:48.385+0300","logger":"stores","caller":"stores/local.go:608","msg":"remove /fc/fast/2/unsealed/s-t0713585-21"}
{"level":"error","ts":"2021-04-30T17:08:48.415+0300","logger":"evtsm","caller":"go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/machine.go:83","msg":"Executing event planner failed: running planner for state Removing failed:\n    github.com/filecoin-project/lotus/extern/storage-sealing.(*Sealing).plan\n        /home/etsvigun/lotus/extern/storage-sealing/fsm.go:250\n  - planner for state Removing received unexpected event sealing.SectorRetryPreCommitWait ({User:{}}):\n    github.com/filecoin-project/lotus/extern/storage-sealing.planOne.func1\n        /home/etsvigun/lotus/extern/storage-sealing/fsm.go:524"}
```

**Comment by SgtCoin** [comment, marked as answer]
Archiving due to age, network is now on v1.13+
